### PR TITLE
chore: [CI-8250]: add tests for CI execution utils

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

add tests for CI execution utils

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by Eric761 — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

File: src/modules/70-pipeline/utils/__tests__/executionUtils.test.ts [hunk 1]
```diff
@@ -639,4 +640,16 @@ describe('ExecutionUtils tests', () => {
...
+
+  describe('Utils for Process CI Data', () => {
+    test('background node map', () => {
+      const newNodeMap = utils.processForCIData({
+        nodeMap: { ...nodeDataBackgroundStep?.data?.executionGraph?.nodeMap },
+        data: nodeDataBackgroundStep
+      })
+      expect(newNodeMap.lYz9fURmRCenQXzLZBCZDQ.startTs).toBe(0)
+      expect(newNodeMap.lYz9fURmRCenQXzLZBCZDQ.endTs).toBe(0)
+      expect(newNodeMap.lYz9fURmRCenQXzLZBCZDQ.status).toBe('Running')
```
Comment (d1wilko) [inline src/modules/70-pipeline/utils/__tests__/executionUtils.test.ts:652]: maybe `lYz9fURmRCenQXzLZBCZDQ` should be exported as a variable from mockExecutionData

File: src/modules/70-pipeline/utils/__tests__/mockJson/mockExecutionData.ts [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+          progressData: undefined,
+          delegateInfoList: [],
+          interruptHistories: [],
+          stepDetails: undefined,
+          strategyMetadata: undefined,
+          executionInputConfigured: false,
+          logBaseKey:
+            'accountId:h61p38AZSV6MzEkpWWBtew/orgId:default/projectId:shauryaTest/pipelineId:BG_step/runSequence:90/level0:pipeline/level1:stages/level2:build/level3:spec/level4:execution/level5:steps/level6:Run_2'
+        },
+        lYz9fURmRCenQXzLZBCZDQ: {
```
Comment (d1wilko) [inline src/modules/70-pipeline/utils/__tests__/mockJson/mockExecutionData.ts:249]: then you could make use of it here like
`[myConst]: {...`

**Review by d1wilko — COMMENTED** [review]
One suggestion for maintainability - but otherwise looking good

File: src/modules/70-pipeline/utils/__tests__/executionUtils.test.ts (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/utils/__tests__/executionUtils.test.ts:652]: sure

**Review by d1wilko — APPROVED** [review]
LGTM

**Review by Eric761 — APPROVED** [review]

**Pull request merged**